### PR TITLE
Explain build and runtime dependencies

### PR DESCRIPTION
**Pull request by caybro**
Explain the Solid build and runtime dependencies in more detail.

Fixes issue #2

### CODE REVIEW

**Review by zeenix — COMMENTED** [review]
Changes are good but commit log can be much better.

**Review by zeenix — CHANGES_REQUESTED** [review]
Sorry i accidently hit the submit button. I was going to say what needs fixing. :)

* "Update" is pretty vague.
* The details section is supposed to be normal English so
  * Capitalize the first e.
  * Finish sentences with '.'.
  * The only valid reasons to break into newline is to start a new para (by leaving a blank line in between) or exceeding the character limit (which is 75). :)

Sorry to point to a GNOME wiki but we have agreed to follow these guidelines: https://wiki.gnome.org/Git/CommitMessages/

**Comment by caybro** [comment]
@zeenix Objections addressed I believe

**Comment by zeenix** [comment]
@caybro Thanks! I wouldn't call them "objections" though. :-)

**Comment by zeenix** [comment]
@caybro The commit remains the same, you probably forgot to push?

File: README.md [hunk 1]
```diff
@@ -7,7 +7,10 @@ Current backends:
 - Solid
 
 ## Dependencies
-- Solid (http://solid.kde.org) + deps: libupnp, flex/bison, udev
+- Solid (https://api.kde.org/frameworks/solid/html/index.html) 
+  - build deps: flex/bison, udev, extra-cmake-modules
+  - manual build: usually just a matter of `sudo apt build-dep solid` for Ubuntu or `sudo dnf builddep kf5-solid` for Fedora
```
Comment (GordanM) [inline README.md]: `sudo apt build-dep solid` doesn't seem to work unless I have populated the sources.list with the location of the solid source code.

```
➜  poky git:(pyro) sudo apt build-dep libkf5solid
Reading package lists... Done
E: You must put some 'source' URIs in your sources.list
```

**Review by GordanM — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (caybro) [reply README.md]: https://wiki.documentfoundation.org/Development/Linux_Build_Dependencies#Debian_.2F_Ubuntu

**Comment by zeenix** [comment]
@caybro Thanks but merges makes things difficult to follow in git history. Please rebase and squash.

**Comment by zeenix** [comment]
oh and you'll need to do a forced push to your branch.

**Comment by GordanM** [comment]
@zeenix I've disabled merge commits for this repository, it should not be possible to do so.

**Comment by rpannek** [comment]
I'd like to link https://pelagicore.github.io/software-factory/chapters/workflow/git-workflow.html here where we explain the workflow with private forks and pull requests, etc.

**Comment by zeenix** [comment]
@rpannek thanks and I'll add a section about recommendation for good commit messages to that chapter too.

**Comment by caybro** [comment]
@zeenix rebased and squashed the commits, should be good now

**Review by rpannek — APPROVED** [review]

**Review by zeenix — APPROVED** [review]

**Pull request merged**